fdbserver/storageserver.actor.cpp

```diff
@@ -848,7 +852,10 @@ struct StorageServer {
 	    fetchKeysParallelismLock(SERVER_KNOBS->FETCH_KEYS_PARALLELISM),
 	    fetchKeysBytesBudget(SERVER_KNOBS->STORAGE_FETCH_BYTES), fetchKeysBudgetUsed(false),
 	    instanceID(deterministicRandom()->randomUniqueID().first()), shuttingDown(false), behind(false),
-	    versionBehind(false), debug_inApplyUpdate(false), debug_lastValidateTime(0), maxQueryQueue(0), counters(this) {
+	    versionBehind(false), debug_inApplyUpdate(false), debug_lastValidateTime(0), maxQueryQueue(0), counters(this),
+	    storageServerEventHolder(makeReference<EventCacheHolder>(ssi.id().toString() + "/BusiestReadTag")) {
+		transactionTagCounter.transactionTagCounterHolder =
+		    makeReference<EventCacheHolder>(ssi.id().toString() + +"/StorageServerSourceTLogID");
```

Comment:
Both of these holders should be used to get the tracking key in the relevant log events. Also, I think you've got them switched.

Like with the cluster controller holder, I would recommend renaming both of these after the trace event being logged.

Also, I would recommend initializing the `TransactionTagCounter`'s holder inside the class. I think you should be able to do this inline with the declaration.